src/extensions/default/HealthData/HealthDataManager.js

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2015 Adobe Systems Incorporated. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+/*jslint vars: true, plusplus: true, devel: true, nomen: true, indent: 4, maxerr: 50 */
+/*global define, $, brackets, console */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    var AppInit             = brackets.getModule("utils/AppInit"),
+        PreferencesManager  = brackets.getModule("preferences/PreferencesManager"),
+        UrlParams           = brackets.getModule("utils/UrlParams").UrlParams,
+        HealthDataUtils     = require("HealthDataUtils"),
+        uuid                = require("thirdparty/uuid");
+
+    var prefs = PreferencesManager.getExtensionPrefs("healthData");
+
+    prefs.definePreference("healthDataTracking", "boolean", true);
+
+    var ONE_DAY = 24 * 60 * 60 * 1000,
+        timeoutVar;
+
+    var params = new UrlParams();
+    params.parse();
+
+    /**
+     * Get the health data which will be sent to the server. Initially it is only one time data.
+     */
+    function getHealthData() {
+        var result = new $.Deferred(),
+            oneTimeHealthData = {};
+
+        var userUuid = PreferencesManager.getViewState("UUID");
+
+        if (!userUuid) {
+            userUuid = uuid.v4();
+            PreferencesManager.setViewState("UUID", userUuid);
+        }
+
+        oneTimeHealthData.uuid = userUuid;
+        oneTimeHealthData.snapshotTime = (new Date()).getTime();
```

Comment:
@peterflynn Thanks. Just compared the performance. Date.now() is slightly better in performance than constructor 'new Date()'